helix-view/src/editor.rs

```diff
@@ -1307,16 +1333,11 @@ impl Editor {
     }
 
     pub async fn wait_event(&mut self) -> EditorEvent {
-        let mut saves: FuturesUnordered<_> = self
-            .documents
-            .values_mut()
-            .map(Document::await_save)
-            .collect();
-
         tokio::select! {
             biased;
 
-            Some(Some(event)) = saves.next() => {
+            Some(event) = self.save_queue.next() => {
+                self.write_count -= 1;
```

Comment:
Should we wait until the event is fully handled before considering the save done?

Reply:
I was considering it, but if there was an error in the event handling that returned and skipped the `write_count` in the future then `flush_writes` would block. We could make the change though

Reply:
Yeah I was just realizing that too. I'm actually thinking that no matter what kind of error we hit, we'd probably always want to decrement this count anyway, so let's leave it.